subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/execution/ResidualProgramCompiler.kt

```diff
@@ -467,6 +467,10 @@ class ResidualProgramCompiler(
         return compileScript(scriptFile, originalScriptPath, scriptDefinition)
     }
 
+    init {
+        org.jetbrains.kotlin.cli.common.environment.setIdeaIoUseFallback()
+    }
```

Comment:
It seems kind of weird that this `init` block is hidden away in the middle of the file instead of at the top.

Also, it seems weird that you are using a constructor to change what seems like some static state about the system.